doc/source/index.rst

```diff
@@ -207,9 +206,8 @@ Getting Involved
 
 .. toctree::
    :maxdepth: -1
-   :caption: Development and Internals
+   :caption: Debugging and Internals
```

Comment:
We can keep this additional, although I would like to make a folder of 

"Debugging and Ray Internal"
internals/index.rst
internals/debugging.rst
internals/profiling.rst
internals/architecture.rst

Reply:
For the most part, it's a code smell to have everything at the top level folder like it would be with code

Reply:
OK, this can be a later PR